mdal/mdal_utils.cpp

```diff
@@ -796,3 +796,70 @@ void MDAL::Error::setDriver( std::string driverName )
 {
   driver = driverName;
 }
+
+void parseDriverFromUri( const std::string &uri, std::string &driver )
+{
+  bool hasDriverSet = ( uri.find( ":\"" ) != std::string::npos );
+  driver = "";
+
+  if ( !hasDriverSet )
+    return;
+
+  driver = MDAL::split( uri, ":\"" )[0];
+}
+
+void parseSpecificMeshFromUri( const std::string &uri, std::string &meshName, int &meshID )
+{
+  bool hasSpecificMeshSet = ( uri.find( "\":" ) != std::string::npos );
+  meshName = "";
+  meshID = 0;
+
+  if ( !hasSpecificMeshSet )
+    return;
+
+  std::vector<std::string> tokens = MDAL::split( uri, "\":" );
+  if ( tokens.size() > 1 )
+  {
+    if ( isdigit( tokens.at( 1 ).c_str()[0] ) ) // number as specific mesh - meshID
+    {
+      char *end;
+      meshID = std::strtol( tokens.at( 1 ).c_str(), &end, 10 );
+    }
+    else // string as specific mesh - meshName
+    {
+      meshName = tokens.at( 1 );
+    }
+  }
+}
+
+void parseMeshFileFromUri( const std::string &uri, std::string &meshFile )
```

Comment:
i think this function can do all three things at once ( parseDriverFromUri and parseSpecificMeshFromUri), it will be better readable